### ISSUE TITLE
Support for brotli

### ISSUE THREAD

**Issue by christophehurpeau** [koajs/compress#21]
Here an interesting link : https://hacks.mozilla.org/2015/11/better-than-gzip-compression-with-brotli/

**Comment by christophehurpeau** [comment]
I could try to make a PR if you're interested

**Comment by jonathanong** [comment]
imo we should just have an api for adding more compression methods

**Comment by coderhaoxin** [comment]
Looks like https://github.com/expressjs/compression/pull/62 ?
This issue is duplicated with #19 ?

**Comment by eknkc** [comment]
BTW, Chrome seems to have brotli enabled by default for upcoming versions:

https://chromium.googlesource.com/chromium/src/+/e3118678848807c0ea60ab086e3d803cb1913098%5E%21/#F0

**Comment by ShimShamSam** [comment]
Any progress on this?

**Comment by jonathanong** [comment]
PR welcomed :)

**Comment by dlebedynskyi** [comment]
Hey, guys, wondering if there is any progress on this? would love to see brotli support added

**Comment by jonathanong** [comment]
was going to implement this but it seems as though brotli is only better on larger sized payloads. we might need to think about how we implement this more.

**Comment by dlebedynskyi** [comment]
@jonathanong followup on that - did some testing myself recently using https://github.com/tuananh/kompression and https://www.npmjs.com/package/shrink-ray

- compression time seems bigger then gzip for around  same dynamic content size of ~15kb 
- brotli does help with static files though roughly 5-10% save on brotli  for static files around ~100kb gzip

**Comment by tracker1** [comment]
Added a PR #80 which implements optional brotli support... I would specify a quality of 5-8 though instead of the default of 11, as it will be much slower than gzip otherwise.  You really don't gain much for dynamic content vs. the extra time for compression.  I did the work and made it optional in case someone *really* wants it, or in case the iltorb library gets significantly better.

my single use case:

```
encoding  level/quality  time   size
gzip      9              36ms   129kb
br        11             441ms  127kb
br        8              43ms   128kb
br        6              38ms   128kb
br        5              36ms   128kb
```

**Comment by syuilo** [comment]
Node.js now supports Brotli natively.

**Comment by dennismphil** [comment]
Any progress on this?

**Comment by indeyets** [comment]
@jonathanong brotli makes a lot of sense for API responses under high load. we usually sticked to `gzip -1` (lowest possible load), but `brotli -3` uses even less cpu resources while providing better compression (even for small files)

CDN doesn't help with API :-/

**Comment by jonathanong** [comment]
closed by #99
